src/contract.rs

```diff
@@ -1207,6 +1163,11 @@ fn execute_match(
         .load(bid_id.as_bytes())
         .map_err(|error| ContractError::LoadOrderFailed { error })?;
 
+    // Validate the requested quote denom in the ask order matches the offered quote denom in the bid order
```

Comment:
ya, it was in the audit. i think in reality, we need to remove the list feature of the quote denom, but since we have support for it right now, this is the quickest "remediation"